### ISSUE TITLE
Downloads not finishing

### ISSUE THREAD

**Issue by SOLIJA** [UnigramDev/Unigram#2617]
**Describe the bug**
When downloading a file, it will get to 99% and then turn back into the download button. I go to the folder where it is downloaded to (the TMP folder) and it shows a file without an extension about the size of the download.

**To Reproduce**
1. Click on Download.
2. Wait for it to Download and it gets to about 99.9%
3. File appears to finish downloading in Unigram, check the folder and it has not.
4. Click on Download again.
5. It goes to start but it shows 99% and then stops again.
6. Download never finishes.

**Version Info**
 - Unigram: 8.4 (6853)
 - Windows: Windows 11 : 21H2 : 22000.376

**Additional context**
Seems to be the same issue as Issue #2420  , but it seems downloads finished for him, but will not for me.

**Comment by Aquathing** [comment]
What kind of file is it? Are you sure your antivirus isn't deleting it?

**Comment by SOLIJA** [comment]
> What kind of file is it? Are you sure your antivirus isn't deleting it?

Can be any file. Tried with antivirus off also.

**Comment by makisukurisu** [comment]
@SOLIJA Have you changed download path?

If no — is TMP folder you are talking about is %LOCALAPPDATA%\Packages\TelegramFZ-LLC.Unigram_1vfw5zm9jmzqy\LocalState\0\temp ?
(Or %LOCALAPPDATA%\Packages\38833FF26BA1D.UnigramPreview_g9c9v27vpyspw\LocalState\0\temp for store release)

**Comment by SOLIJA** [comment]
> 

FrayxRulez, I followed the directions on your link and sent it to you in Unigram.

**Comment by RathHunt** [comment]
I have been also facing this issue but on windows executables ( exe , msi , bat ) , not only while downloading but also sending results in failed

**Comment by SOLIJA** [comment]
> 

Not sure if you have the Windows store version or not, but I did and went into the App settings and just reset the whole app. Now it seems to be working for me.

I sent him my TDLib-log and his suggestion was to reset default download location. Which I had the problem before switching it to a different location. So, I went to go reset to the default location, and it wouldn't let me reset it. Which prompted the app reset I mentioned.
